cmd/dogstatsd/main.go

```diff
@@ -73,26 +72,30 @@ func init() {
 	dogstatsdCmd.AddCommand(versionCmd)
 
 	// local flags
-	startCmd.Flags().StringVarP(&confPath, "cfgpath", "c", "", "path to datadog.yaml")
+	startCmd.Flags().StringVarP(&confPath, "cfgpath", "c", "", "path to folder containing dogstatsd.yaml")
 	config.Datadog.BindPFlag("conf_path", startCmd.Flags().Lookup("cfgpath"))
 	startCmd.Flags().StringVarP(&socketPath, "socket", "s", "", "listen to this socket instead of UDP")
 	config.Datadog.BindPFlag("dogstatsd_socket", startCmd.Flags().Lookup("socket"))
 }
 
 func start(cmd *cobra.Command, args []string) error {
-	confFilePath := config.Datadog.GetString("conf_path")
-	if len(confFilePath) != 0 {
-		// if the configuration file path was supplied on the command line,
-		// add that first so it's first in line
-		config.Datadog.AddConfigPath(confFilePath)
-		// If they set a config file directly, let's try to honor that
-		if strings.HasSuffix(confFilePath, ".yaml") {
-			config.Datadog.SetConfigFile(confFilePath)
+	configFound := false
+
+	// a path to the folder containing the config file was passed
+	if len(confPath) != 0 {
+		// we'll search for a config file named `dogstastd.yaml`
+		config.Datadog.SetConfigName("dogstatsd")
```

Comment:
What do you think about falling back on `datadog.yaml` and trying that as well? I feel like this will be a common mistake.